### PR TITLE
HPCC-25080 Add __IS_LITTLE_ENDIAN__ constant to ECL language

### PR DESCRIPTION
**Pull request by dcamper**
Signed-off-by: Dan S. Camper <dan.camper@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [x] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:

Manual testing.

### CODE REVIEW

**Comment by dcamper** [comment]
@ghalliday Please review.  Thanks!

**Comment by ghalliday** [comment]
@dcamper although it is only a small change I'm not quite sure about it.

a) I think you can already swap bits safely with (unsigned8)TRANSFER((LITTLE_ENDIAN unsigned8)x, BIG_ENDIAN UNSIGNED8)

b) I was expecting __IS_LITTLE_ENDIAN__(type) rather than a flag for the system.  I suspect either implementation is fine though.

c) Are we ever going to run on a big endian system?  It seems little endian is now the defacto standard.

I'll think some more.

**Comment by dcamper** [comment]
@ghalliday Thanks for taking a look.  You have a good point with (c) because, if LE is indeed the standard then everything else is moot.  If it is not standard, I was going after a simple compile-time constant that could be used for conditionally including code.  (a) works without the compile-time test but unless it has its own BE/LE tests -- and it may, I haven't checkd -- it will unnecessarily expensive.

Bottom line, please feel _entirely_ free to reject the Jira and PR.  The whole thing originated from logically considering a coding scenario, not from any actual user-provided problem.

**Comment by dcamper** [comment]
I should also point out that (a) is a "force the integer to BE" but the purpose of my addition is to allow the ECL programmer to "switch byte order to the opposite of native order" using the explicit transfers.  That is a bit different.  But as before, if (c) holds then this whole thing should go away.

**Comment by ghalliday** [comment]
(a)Should swap the byte order whatever the underlying representation.
Closing because I'm not convinced it is needed, or the best way to provide the functionality.  (Sorry!)
